### PR TITLE
GH-37510: [C++] Don't install bundled Azure SDK for C++

### LINKED ISSUE #37510: [C++] Bundled Azure SDK for C++ should not be installed
### Describe the enhancement requested

`libazure-*.a` such as `libazure-core.a` should not be installed. They must be included in `libarrow_bundled_dependencies.a`.

### Component(s)

C++

### PR DESCRIPTION
**Pull request by kou**
### Rationale for this change

It's an internal bundled library. We should not install it as a part of Arrow.

### What changes are included in this PR?

Exclude all Azure SDK for C++ jobs including install jobs aren't executed by default. Building jobs are executed because they are required to build Arrow.

### Are these changes tested?

Yes.

### Are there any user-facing changes?

Yes.
* Closes: #37510

### CODE REVIEW

File: cpp/cmake_modules/ThirdpartyToolchain.cmake [hunk 1]
```diff
@@ -5070,8 +5070,20 @@ endif()
...
+    #
+    # At least CMake 3.22 on Ubuntu 22.04 works. So we use 3.22
+    # here. We may be able to use more earlier version here.
+    message(FATAL_ERROR "Building Azure SDK for C++ requires at least CMake 3.22. "
+      "(At least we can't use CMake 3.16)")
+  endif()
   message(STATUS "Building Azure SDK for C++ from source")
   fetchcontent_declare(azure_sdk
+                       # EXCLUDE_FROM_ALL is available since CMake 3.28
+                       # EXCLUDE_FROM_ALL TRUE
```
Comment (assignUser) [inline cpp/cmake_modules/ThirdpartyToolchain.cmake:5086]: iirc you could just leave that there and cmake will ignore it if it doesn't know it

**Review by assignUser — APPROVED** [review]
With the release imminent and there being a clear workaroudn (system azure sdk) I am ok with this solution. 

A future improvement might be installing the sdk into the build tree instead of the host system by modifing:
```
        RUNTIME DESTINATION ${CMAKE_INSTALL_BINDIR} # DLLs (if produced by build) go to "/bin"
        LIBRARY DESTINATION ${CMAKE_INSTALL_LIBDIR} # static .lib files
        ARCHIVE DESTINATION ${CMAKE_INSTALL_LIBDIR} # .lib files for DLL build
        INCLUDES DESTINATION ${CMAKE_INSTALL_INCLUDEDIR}
```
Or setting EXCLUDE_FROM_ALL for each of the azure targets might also work? Not sure if dir scope is needed to supress the installation...

File: cpp/cmake_modules/ThirdpartyToolchain.cmake (hunk 1 above)
Comment (kou) [reply cpp/cmake_modules/ThirdpartyToolchain.cmake:5086]: Unfortunately, CMake reports an error for this case...

**Comment by kou** [comment]
> A future improvement might be installing the sdk into the build tree instead of the host system by modifing:

Does this approach apply a patch to Azure SDK for C++? I want to avoid having a patch...

> Or setting EXCLUDE_FROM_ALL for each of the azure targets might also work? Not sure if dir scope is needed to suppress the installation...

It will work but I want to maintain target list for Azure SDK for C++... It'll increase maintenance cost like Abseil...

**Review by felipecrv — APPROVED** [review]

**Comment by kou** [comment]
I'll merge this for 14.0.0.

**Pull request merged**